File: markdown/private/core/validate_input_metadata.py
import argparse
import json

from markdown.private.utils.metadata import InputMetadata


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("in_file")
    parser.add_argument("out_file")
    args = parser.parse_args()

    with open(args.in_file, encoding="utf-8") as f:
        metadata = InputMetadata.model_validate_json(f.read())

    with open(args.out_file, mode="w", encoding="utf-8") as f:
        json.dump(
            metadata.model_dump(
                mode="json",
                by_alias=True,
                exclude_unset=True,
                exclude_defaults=True,
            ),
            f,
            sort_keys=True,
            indent=4,
        )


if __name__ == "__main__":
    main()

File: markdown/private/core/validate_input_metadata.rs
use clap::Parser;
use markdown::json::{from_str, Json};
use markdown::metadata::InputMetadata;
use std::error::Error;
use std::fs::read_to_string;

#[derive(Parser)]
#[command(version, about, long_about = None)]
struct Cli {
    in_file: String,
    out_file: String,
}

fn main() -> Result<(), Box<dyn Error>> {
    let args = Cli::parse();

    let data = read_to_string(args.in_file)?;
    let metadata: InputMetadata = from_str(&data)?;

    metadata.write(args.out_file)
}
